pkg/reconciler/revision/resources/pa.go

```diff
@@ -49,6 +49,19 @@ func MakePA(rev *v1alpha1.Revision) *av1alpha1.PodAutoscaler {
 				Name:       names.Deployment(rev),
 			},
 			ProtocolType: rev.GetProtocol(),
+			Reachability: func() av1alpha1.ReachabilityType {
```

Comment:
This is fine for me in general but we need to make sure subsequent PRs land in this version or we'll be in the business of having to support backwards compat cases in this logic from the getgo should we want to change it. I think it currently doesn't reflect the latest state we discussed in the meeting, which is fine for now. Please beware of the compat implications though.

Reply:
Hey @markusthoemmes, The plan is to get #5147 in this release, but this change is specifically designed to have zero impact if it doesn't land, and be future compatible with #5147 when it does land. If I'm missing something, I definitely want to address it.

Can you elaborate on these? 

>  support backwards compat cases in this logic from the getgo should we want to change it

Which cases? 

>  I think it currently doesn't reflect the latest state we discussed in the meeting

The four states we talked about were:
* `nil` (now `ReachabilityUnknown`) for backwards compatibility
* `true` (now `ReachabilityReachable`)
* `false` (now `ReachabilityUnreachable`)
*  activating (treated as `ReachabilityUnreachable` before but [should have been](https://github.com/knative/serving/pull/4884#discussion_r314121707) `ReachabilityReachable`, now `ReachabilityUnknown`)

Are there others?

Reply:
Though about it over chopping veggies: We're indeed good here I think. This logic is simple enough to change up at a later stage.